src/univers/gem.py

```diff
@@ -242,6 +237,26 @@ def __gt__(self, other):
     def __ge__(self, other):
         return self.__cmp__(other) >= 0
 
+    @property
+    def segments(self):
+        """
+        Yield segments for this version where segments are
+        ints or strings parsed from the original version string.
+        """
+        find_segments = re.compile(r"[0-9]+|[a-z]+", re.IGNORECASE).findall
+        for seg in find_segments(self.version):
+            yield int(seg) if seg.isdigit() else seg
+
+    @property
+    def canonical_segments(self):
+        """
+        Yield  "canonical segments" for this version using
+        the Rubygems way for canonicalization.
+        """
+        for segments in self.split_segments():
+            segs = list(dropwhile(lambda s: s == 0, reversed(segments)))
```

Comment:
I reckon you inherited from this but this is a rather complex expression IMHO... is there a way to rewrite this that would be easier to read?
Or at least some comment that explain what this is doing?

Reply:
It's an original adoption from RubyGems [`version.rb`](https://github.com/ruby/ruby/blob/13305bf0c9a03f24b5ab667784ff1dc1f10b3d9f/lib/rubygems/version.rb#L374-L379)
```rb
  def canonical_segments
    @canonical_segments ||=
      _split_segments.map! do |segments|
        segments.reverse_each.drop_while {|s| s == 0 }.reverse
      end.reduce(&:concat)
  end
```

Will add comments there and make it return a tuple instead of generator